feat(docs): adding reindex docs following #2142 (#1023)
questdb/questdb.io: docs/reference/sql/reindex.md

---
title: REINDEX
sidebar_label: REINDEX
description: REINDEX SQL keyword reference documentation.
---

> Rebuilds one or more index columns.

Reviewer: nit:
```suggestion
Rebuilds one or more index columns of the given table.
```

> This operation is intended to be used after a QuestDB crash, when the index data are corrupted and the table cannot be opened for writes.

Reviewer: nit: I'd avoid "QuestDB" crash:
```suggestion
This operation is intended to be used after a hardware or software crash, when the index data are corrupted and the table cannot be opened for writes.
```

`REINDEX` rebuilds index non-transitionally. The operation can only be performed when there is no other reader and writer working on the table. During the operation, the table is locked and no read and write should be performed on the selected table.

:::info

For more information about indexes please refer to the
[INDEX documentation](/docs/concept/indexes)

:::

## Syntax

![Flow chart showing the syntax of the REINDEX keyword](/img/docs/diagrams/reindex.svg)

## Options

> By default, `REINDEX` rebuilds all the columns that have been indexed in the selected table. The following options can be used to narrow down the scope of the operation:

Reviewer: ```suggestion
By default, `REINDEX` rebuilds all indexes in the selected table. The following options can be used to narrow down the scope of the operation:
```

- `COLUMN`: When defined, `REINDEX` rebuilds the index for the selected column. 

> - `PARTITION`: When defined, `REINDEX` rebuilds index for all the columns that have been indexed in the selected partition. The partition name must match the name of the directory for the given partition. The naming convention is detailed in [Partitions](docs/concept/partitions)

Reviewer: ```suggestion
- `PARTITION`: When defined, `REINDEX` rebuilds index files in the selected partition only. The partition name must match the name of the directory for the given partition. The naming convention is detailed in [Partitions](docs/concept/partitions).
```

## Example

> ```questdb-sql title="Rebuilding an index"

Reviewer: nit: I'd add a few more examples: rebuild all indexes in a table, rebuild all indexes in a partition

Author: Added more example with explanation.

> 

Reviewer: Redundant newline here.

> REINDEX TABLE trades COLUMN instruments PARTITION  '2021-12-17' LOCK EXCLUSIVE

Reviewer: nit:
```suggestion
REINDEX TABLE trades COLUMN instruments PARTITION  '2021-12-17' LOCK EXCLUSIVE;
```

> 

Reviewer: Redundant newline here.
